miner/worker.go

```diff
@@ -1403,6 +1403,13 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 	if w.chainConfig.DAOForkSupport && w.chainConfig.DAOForkBlock != nil && w.chainConfig.DAOForkBlock.Cmp(header.Number) == 0 {
 		misc.ApplyDAOHardFork(env.state)
 	}
+	if w.chainConfig.CurieBlock != nil && w.chainConfig.CurieBlock.Cmp(header.Number) == 0 {
+		misc.ApplyCurieHardFork(env.state)
+
+		// zkEVM requirement: Curie transition block has 0 transactions
+		w.commit(nil, w.fullTaskHook, true, tstart)
```

Comment:
why does the transition block need to be empty?

Reply:
Context: https://scrollco.slack.com/archives/C02PKAQ2430/p1715917234850479?thread_ts=1715690937.472719&cid=C02PKAQ2430

Reply:
thanks.